Fix @profview to correctly include interactive threads

jaredjeya opened the pull request:
@profview does not show information from all worker threads because it fails to count the total number of threads correctly, incorrectly only counting the number of worker threads. However, since Julia orders interactive threads before worker threads, we miss out on the last worker thread (assuming one interactive thread). Since Julia will by default have 1 interactive thread starting with v1.12, it's important to fix this before that releases.

Fixes #3820.

I've made the minimal possible change here and tested that this works on my own machine. I've also made a small GUI change: the dropdown on the profiler now lists "All", "Interactive: x", "Worker: y" instead of just numbering the threads. I've made the choice that worker threads start at 1, instead of `nthreads(:interactive) + 1`, but this can of course be changed.

- [X] End-user documentation check. If this PR requires end-user documentation, please add that at https://github.com/julia-vscode/docs.

I don't think this requires a documentation update, it should be self-explanatory.

jaredjeya commented:
I've updated this to make no assumptions about thread ordering and to be a little more robust. Note also that I've not included the foreign threadpool (this was never included, to be clear). In the future there could be a keyword argument to the macro that decides whether to also include foreign threads?

Also, it seems like changing the name "all" to "All threads" (or in fact, anything but "all") means that option gets sorted last in the profile viewer.  That's particularly annoying as it no longer gets selected by default. I'm going to assume there's some code in there that grabs it and puts it at the top? So you might want to change it back to "all" before merging this - OR, change the code in whatever actually displays the profile to match. Unfortunately I can't figure out where that is!

pfitzseb commented:
> Also, it seems like changing the name "all" to "All threads" (or in fact, anything but "all") means that option gets sorted last in the profile viewer. That's particularly annoying as it no longer gets selected by default. I'm going to assume there's some code in there that grabs it and puts it at the top?

That's at https://github.com/pfitzseb/jl-profile.js/blob/a13f2ef7852bc1782ec180e3efd4ccef2da7ba6d/profile-viewer.ts#L146C17-L151. Would be nice if you could just change it back to `all` for now so we can get the PR merged sooner rather than later.

jaredjeya commented:
Ah, well that explains why nothing I changed in `src/interactive/profiler.ts` made any difference to which one got selected! I've changed it back now.
